nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/CaptureBoxEvents.java

```diff
@@ -0,0 +1,227 @@
+                }
+            }
+
+            @Override
+            public void onNextPosition(long position) {
+                getLogger().debug("Next position: {}", position);
+            }
+
+            @Override
+            public boolean onException(Throwable e) {
+                getLogger().warn("An error has been received from the stream.", e);
+                return true;
+            }
+        });
+
+        eventStream.start();
+    }
+
+    @OnStopped
+    public void stopped() {
+        if (eventStream != null && eventStream.isStarted()) {
+            eventStream.stop();
+        }
+    }
+
+    @Override
+    public List<ConfigVerificationResult> verify(ProcessContext context, ComponentLog verificationLogger, Map<String, String> attributes) {
+
+        final List<ConfigVerificationResult> results = new ArrayList<>();
+        BoxClientService boxClientService = context.getProperty(BoxClientService.BOX_CLIENT_SERVICE).asControllerService(BoxClientService.class);
+        boxAPIConnection = boxClientService.getBoxApiConnection();
+
+        try {
+            boxAPIConnection.refresh();
+            results.add(new ConfigVerificationResult.Builder()
+                    .verificationStepName("Box API Connection")
+                    .outcome(Outcome.SUCCESSFUL)
+                    .explanation("Successfully validated Box connection")
+                    .build());
+        } catch (Exception e) {
+            getLogger().warn("Failed to verify configuration.", e);
+            results.add(new ConfigVerificationResult.Builder()
+                    .verificationStepName("Box API Connection")
+                    .outcome(Outcome.FAILED)
+                    .explanation(String.format("Failed to validate Box connection: %s", e.getMessage()))
+                    .build());
+        }
+
+        return results;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        if (events.isEmpty()) {
+            context.yield();
+            return;
+        }
+
+        final FlowFile flowFile = session.create();
+        final List<BoxEvent> eventList = new ArrayList<>();
+        final int nbEvents = events.drainTo(eventList);
+
+        try {
+            // converting events to JSON using the library coming with Box SDK
+            final JsonArray jsonEvents = Json.array();
+            for (BoxEvent event : eventList) {
+                jsonEvents.add(toRecord(event));
+            }
+
+            try (final OutputStream out = session.write(flowFile)) {
+                final Writer writer = new OutputStreamWriter(out, StandardCharsets.UTF_8);
+                jsonEvents.writeTo(writer);
+                writer.flush();
+            }
+        } catch (Exception e) {
+            getLogger().error("Failed to write events to FlowFile due to {}; will re-queue events and try again", e.getMessage(), e);
```

Comment:
Recommend avoiding the `due to {}` approach since the message is already included in the exception stack trace.